wandb/filesync/dir_watcher.py

```diff
@@ -120,9 +120,10 @@ def last_uploaded(self):
             if time_elapsed < self.RATE_LIMIT_SECONDS:
                 return time_elapsed
             # Check rate limit by size increase
-            size_increase = self.current_size / float(self._last_uploaded_size)
-            if size_increase < self.RATE_LIMIT_SIZE_INCREASE:
-                return time_elapsed
+            if float(self._last_uploaded_size) > 0:
+                size_increase = self.current_size / float(self._last_uploaded_size)
+                if size_increase < self.RATE_LIMIT_SIZE_INCREASE:
+                    return time_elapsed
```

Comment:
I wonder how is `os.path.getsize(self.file_path)` ever a negative number?  What does that even mean?

Reply:
I am not sure that will ever be a case - at least not intentionally. Stackoverflow seems to suggest that there might be the rare case of integer overfow depending on your Python config (https://stackoverflow.com/questions/5167120/why-does-os-path-getsize-return-a-negative-number-for-a-10gb-file). That being said, the intention here is to protect against the 0 case. Would prefer a not equal here rather than the inequality?

Reply:
Ahh I see, no this is fine.  We should verify returning 0 here doesn't make us attempt to upload an empty file... I vaguely recall having logic in the uploader that prevents that.

Reply:
Makes sense - thanks for calling that out. As I am taking a closer look at this code, the condition to save a file when on_modified is called seems quite particular. I think it effectively resolves down to the following:
```
(
    "the file has never been synced" OR
    "the file has been modified since last sync" OR
    (
        "the file size has grown by less than 20% since last sync" AND
        "the time since last sync is greater than 60-to-1200 seconds depending on size of file"
    )
) 
```
I am not sure how we can ever be in the third case of the OR statement above (how can the file NOT be updated since the last sync, yet change in size?). Also, notice that as implemented, the `force` parameter nor the `RATE_LIMIT_SECONDS` attribute have any effect on the condition.

It seems we might want to revise this condition (maybe not?). Spitballing here, how does this sound: save file when:
```
(
    "force is true" OR
    "the file has never been synced" OR
    (
        "the file has been modified since last sync" AND
        (
            "the time since last sync is greater than 60-to-1200 seconds depending on size of file" OR
            (
                "the file size has CHANGED by MORE than 20% since last sync" AND
                "the time since last sync is greater than RATE_LIMIT_SECONDS(15)"
        )
    )
)
``` 

LMK your thoughts!

Reply:
I'm having trouble getting into a deep nested logic state of mind...  I'm thinking we just keep this as it is for now and spend our mental energy more exciting and impactful issues 😉